### PR TITLE
create a component to handle the search bar with debounce

### PR DESCRIPTION
**Pull request by Nexucis**
Signed-off-by: Augustin Husson <husson.augustin@gmail.com>

### CODE REVIEW

**Comment by Nexucis** [comment]
I'm wondering if we want to have this behavior directly in the v2.33. WDYT @juliusv ?

File: web/ui/react-app/src/pages/serviceDiscovery/Services.tsx [hunk 1]
```diff
@@ -117,12 +116,7 @@ export const ServiceDiscoveryContent: FC<ServiceMap> = ({ activeTargets, dropped
     <>
       <h2>Service Discovery</h2>
       <Container>
-        <InputGroup>
-          <InputGroupAddon addonType="prepend">
-            <InputGroupText>{<FontAwesomeIcon icon={faSearch} />}</InputGroupText>
-          </InputGroupAddon>
-          <Input autoFocus onChange={handleSearchChange} placeholder="Filter by labels" />
-        </InputGroup>
+        <SearchBar handleChange={handleSearchChange} placeholder="filter by labels" />
```
Comment (juliusv) [inline web/ui/react-app/src/pages/serviceDiscovery/Services.tsx:119]:
```suggestion
        <SearchBar handleChange={handleSearchChange} placeholder="Filter by labels" />
```

**Comment by juliusv** [comment]
> I'm wondering if we want to have this behavior directly in the v2.33.

I'm fine either way. Most pages will be small enough for the lack of debounce to cause a problem, and it's an optional new functionality.

@beorn7 do you already know if you're going to do an rc.1 anyway? Then maybe we could include this.

**Comment by juliusv** [comment]
:+1: from a code perspective

**Comment by beorn7** [comment]
Yes, I think we should have this in 2.33.0. @Nexucis could you change this PR to be on top of the `release-2.33` branch?

**Comment by Nexucis** [comment]
I changed the based branch @beorn7 !

**Review by beorn7 — APPROVED** [review]

**Pull request merged**